src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_LENGTH];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        arrayIncrease(size + 1);
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        arrayIncrease(size + 1);
+        if (checkIndex(index) || index == size) {
+            System.arraycopy(data, index, data, index + 1, size - index);
+            data[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is not exists");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int temp = size;
+        for (int i = 0; i < list.size(); i++) {
+            arrayIncrease(size + 1);
```

Comment:
You are not correctly resizing the array for the `addAll` method. You should calculate the total required size first and then resize if necessary.